feat: support filtering users table by login type (#17238)
coder/coder: docs/admin/users/index.md

  `status:active last_seen_before:"2023-07-01T00:00:00Z"`
- To find users who were created between January 1 and January 18, 2023:
  `created_before:"2023-01-18T00:00:00Z" created_after:"2023-01-01T23:59:59Z"`

> - To find users who have login type as github and is a member:
>     `login_type:github role:member`

Reviewer: ```suggestion
- To find users who log in using GitHub:
    `login_type:github`
```
`role:member` shows everyone, as everyone is a member.

  the RFC3339Nano format.
- `created_before` and `created_after` - The time a user was created. Uses the
  RFC3339Nano format.

> - `login_type` - Represents the login type of the user. Refer here for all the roles [LoginType documentation](https://pkg.go.dev/github.com/coder/coder/v2/codersdk#LoginType)

Reviewer: ```suggestion
- `login_type` - Represents the login type of the user. Refer to the [LoginType documentation](https://pkg.go.dev/github.com/coder/coder/v2/codersdk#LoginType) for a list of supported values.
```
